dockerfile-image-update-cmdtool/src/test/java/com/salesforce/dockerfileimageupdate/subcommands/impl/AllTest.java

```diff
@@ -0,0 +1,172 @@
+        Mockito.verify(dockerfileGithubUtil, times(0)).createPullReq(eq(parentRepo), eq("branch"),
+                eq(retrievedRepo), anyString());
+    }
+
+    @Test
+    public void testChangeDockerfiles_pullRequestCreation() throws Exception {
+        DockerfileGithubUtil dockerfileGithubUtil = mock(DockerfileGithubUtil.class);
+
+        GHRepository initialRepo = mock(GHRepository.class);
+        when(initialRepo.isFork()).thenReturn(true);
+        when(initialRepo.getFullName()).thenReturn("repo");
+
+        GHRepository retrievedRepo = mock(GHRepository.class);
+        GHRepository parentRepo = mock(GHRepository.class);
+        when(parentRepo.getFullName()).thenReturn("test2");
+        when(retrievedRepo.getParent()).thenReturn(parentRepo);
+        when(retrievedRepo.getDefaultBranch()).thenReturn("branch");
+
+        when(dockerfileGithubUtil.getRepo(initialRepo.getFullName())).thenReturn(retrievedRepo);
+
+        Map<String, String> parentToPath = ImmutableMap.of(
+                "test1", "test",
+                "test2", "correct",
+                "test3", "test",
+                "test4", "test");
+
+        Map<String, String> parentToImage = ImmutableMap.of(
+                "test1", "image1",
+                "test2", "image2",
+                "test3", "image3",
+                "test4", "image4");
+
+        Map<String, String> imageToTagMap = ImmutableMap.of(
+                "image1", "tag1",
+                "image2", "tag2",
+                "image3", "tag3",
+                "image4", "tag4");
+
+        All all = new All();
+        all.loadDockerfileGithubUtil(dockerfileGithubUtil);
+        all.changeDockerfiles(parentToPath, parentToImage, imageToTagMap,
+                initialRepo, "Automatic Dockerfile Image Updater", "");
+
+        Mockito.verify(dockerfileGithubUtil, times(1)).getRepo(anyString());
+        Mockito.verify(dockerfileGithubUtil, times(1)).tryRetrievingContent(eq(retrievedRepo),
+                eq("correct"), eq("branch"));
+        Mockito.verify(dockerfileGithubUtil, times(1))
+                .modifyOnGithub(any(), eq("branch"), eq("image2"), eq("tag2"), anyString());
+        Mockito.verify(dockerfileGithubUtil, times(1)).createPullReq(eq(parentRepo), eq("branch"),
+                eq(retrievedRepo), anyString());
+    }
+
+    @Test
+    public void testParseStoreToImagesMap() throws Exception {
+        DockerfileGithubUtil dockerfileGithubUtil = mock(DockerfileGithubUtil.class);
+        when(dockerfileGithubUtil.getMyself()).thenReturn(mock(GHMyself.class));
+        when(dockerfileGithubUtil.getRepo(anyString())).thenReturn(mock(GHRepository.class));
+        GHContent mockContent = mock(GHContent.class);
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        when(mockContent.read()).thenReturn(classloader.getResourceAsStream("image-store-sample.json"));
```

Comment:
Any reason this couldn't be inline instead of jumping to files?

Reply:
I felt this would be more readable and there wouldn't be any json embedded in a string or something like that.

Reply:
The things I generally don't like about this approach are:
* You get a filesystem involved
* You have to switch contexts to figure out what's in the file rather than stay within the same test

Reply:
Feel free to send a PR to improve it ;)